File: socialuni-api/src/main/java/com/socialuni/api/feignAPI/SocialuniLocationAPI.java
package com.socialuni.api.feignAPI;

import com.socialuni.api.model.QO.location.LocationQueryQO;
import com.socialuni.api.model.QO.location.LocationQueryRO;
import com.socialuni.social.api.model.ResultRO;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.validation.Valid;

@RequestMapping("location")
@FeignClient(name = "location", url = "${socialuni.server-url:https://api.socialuni.cn}")
public interface SocialuniLocationAPI {
    @PostMapping("queryLocation")
    ResultRO<LocationQueryRO> queryLocation(@RequestBody @Valid LocationQueryQO queryQO);
}

File: socialuni-js/packages/socialuni-api/src/api/socialuni/SocialuniLocationAPI.ts
import {QQMapResult} from "../../model/location/QQMapResult";
import LocationQueryQO from "../../model/location/LocationQueryQO";
import ResultRO from "../../model/social/ResultRO";
import request from '../../request/request';

export default class SocialuniLocationAPI {
    static queryLocation(location: LocationQueryQO = new LocationQueryQO()): Promise<ResultRO<QQMapResult>> {
        //小程序，且注册了qq的地图key
        return request.post<QQMapResult>('socialuni/location/queryLocation', location) as any
    }
}
